va_explorer/users/utils/field_worker_linking

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+"""
+Created on Mon Apr 12 14:33:57 2021
+
+@author: babraham
+"""
+from django.db.models import F
+
+from va_explorer.users.models import User
+from va_explorer.va_data_management.models import VerbalAutopsy, VaUsername
+from va_explorer.va_data_management.utils.location_assignment import fuzzy_match
+
+import pandas as pd
+import re
+from numpy import nan
+
+
+# get table with basic info for all users in system. No PII included in result
+def get_anonymized_user_info(): 
+    # export user data in way that is consistent with user form 
+    # get user form fields
+    form_fields = get_form_fields(orient="h")
+    # figure out which fields are permissions. Assumes that all boolean fields are permissions
+    permissions = form_fields.query("type=='BooleanField'").index.tolist()
+    user_data = User.objects\
+    .select_related('location_restrictions')\
+    .select_related('groups')\
+    .select_related('user_permissions')\
+    .values(
+            'uuid',
+            'is_active',
+            'is_staff',
+            role = F('groups__name'),
+            locations = F('location_restrictions__name'), 
+            permissions = F('user_permissions__codename')         
+    )
+    user_df = pd.DataFrame.from_records(user_data).rename(columns={'locations': 'location_restrictions'})
+    user_perms = (user_df[['uuid', 'permissions']]
+        .pivot_table(index='uuid', columns='permissions', values='uuid', aggfunc=lambda x: len(x) > 0)
+        .fillna(False)
+        .reset_index()
+        )
+    user_df = user_df.drop(columns="permissions").drop_duplicates().merge(user_perms, how="left")
+
+# update field worker usernames by matching names against all known interviewer names from VAs
+def link_fieldworkers_to_vas(emails=None, debug=False, match_threshold=80):
```

Comment:
Since this is a new file, it's a bit hard to tell what changed exactly - were the `match_threshold` parameters raised as discussed or did that end up not being necessary?

Reply:
I ended bumping it up from 75% to 80%. This will result in slightly more conservative matching